File: functions/node/models/product.js
const mongoose = require('mongoose')



//new class
const productSchema = new mongoose.Schema({
    artist:{
        type: String,
    },
    price:{
        type: Number,
    },
    type:{
        type: String,
    },
    urls:{
        type: Array,
    },
    description:{
        type: String,
    },
})

module.exports = mongoose.model('Product', productSchema);

File: lib/models/product.dart
class ProductModel {
  String? artist;
  int? price;
  String? type;
  String? description;
  List? urls;

  ProductModel(
    this.artist,
    this.price,
    this.type,
    this.description,
    this.urls,
  );

  ProductModel.fromJson(
    Map<String, dynamic> prod,
  ) {
    artist = prod['artist'];
    price = prod['price'] ?? 0;
    type = prod["type"] ?? 'empty';
    description = prod["description"] ?? 'Empty';
    urls = prod["urls"] ?? [];
  }

  Map<String, dynamic> toJson() => {
        'artist': artist,
        'price': price,
        'type': type,
        'description': description,
        'urls': urls,
      };
}
